middlewares/auth.middleware.js

```diff
@@ -0,0 +1,40 @@
+exports.isAuthenticated = (req, res, next) => {
+    if (req.user) {
+        return next()
+    } else {
+        res.render('../views/index')
+    }
+}
+
+exports.isNotAuthenticated = (req, res, next) => {
+    if (req.user) {
+        res.redirect('/')
+    } else {
+        return next()
+    }
+}
+
+exports.isadmin=(req,res,next) =>{
```

Comment:
Check naming convention. Should be `is_authenticated`, `is_admin`, etc. I'd suggest removing `is_authenticated` and checking the everything here. Why is `is_not_authenticated` a part?